packages/@aws-cdk-containers/ecs-service-extensions/test/integ.datadog-agent.ts

```diff
@@ -0,0 +1,104 @@
+import { Mesh } from '@aws-cdk/aws-appmesh';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as secretsManager from '@aws-cdk/aws-secretsmanager';
+import { App, Stack } from '@aws-cdk/core';
+import { AppMeshExtension, DatadogAgent, Container, Environment, HttpLoadBalancerExtension, Service, ServiceDescription } from '../lib';
+
+const app = new App();
+const stack = new Stack(app, 'aws-ecs-integ');
+const mesh = new Mesh(stack, 'my-mesh');
+const environment = new Environment(stack, 'production');
+
+// Substitute in your own secret when running the integration test
+const SECRET_ARN = 'arn:aws:secretsmanager:us-west-2:209640446841:secret:datadog-api-key-secret-JxST1g';
+
+// Import a secret from the account.
+const datadogApiKey = secretsManager.Secret.fromSecretAttributes(stack, 'datadog-api-key-secret', {
+  secretCompleteArn: SECRET_ARN,
+});
+
+/** Name service */
+const nameDescription = new ServiceDescription();
```

Comment:
I wonder if this is the right place for this service. Would it make more sense to add to the cdk samples repo instead?

Reply:
My goal right now is to collect a large number of extensions into one package to start with, for the network adoption effect. Once CDK goes 2.0 we can look into decoupling into "packages" with the major version increment. Open to discussing alternative approaches though